File: src/api/predict.js
import { execSync } from "child_process";
export default function handler(req, res) {
  const { temp, hum, co2 } = req.body;
  // panggil Python script yang load .joblib dan predict
  const result = execSync(`python3 predict.py ${temp} ${hum} ${co2}`)
    .toString()
    .trim();
  res.json({ label: parseInt(result) });
}

File: src/api/predict.py
import sys, joblib
model = joblib.load("public/model_random_forest.joblib")
temp, hum, co2 = float(sys.argv[1]), float(sys.argv[2]), float(sys.argv[3])
label = model.predict([[temp, hum, co2]])[0]
print(int(label))